### PR TITLE
add infrastructure-ipam-in-cluster parts to include it during management cluster upgrades

### PR DESCRIPTION
**Pull request by chrischdi**
### Which issue(s) this PR fixes

Fixes the upgrade mechanism to also upgrade the ipam provider when it is installed.

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by christianang** [comment]
Can you rebase this, since this was merged: https://github.com/vmware-tanzu/tanzu-framework/pull/4529

**Comment by chrischdi** [comment]
Thanks @christianang , squashed + rebased.

File: tkg/fakes/config/bom/tkg-bom-v1.3.1.yaml [hunk 1]
```diff
@@ -89,6 +89,12 @@ components:
       capzControllerImage:
         imagePath: cluster-api/cluster-api-azure-controller
         tag: v0.4.8-47-gfbb2d55b_vmware.1
+  cluster-api-ipam-provider-in-cluster:
```
Comment (christianang) [inline tkg/fakes/config/bom/tkg-bom-v1.3.1.yaml:92]: This section is actually in the file already via https://github.com/vmware-tanzu/tanzu-framework/pull/4529 a couple of lines up. So you should be able to remove this section.

**Comment by chrischdi** [comment]
Tested downstream via upgrade-vc7: it is now able to upgrade the ipam provider and also makes use of the updated manifests.

**Review by christianang — APPROVED** [review]

**Review by lubronzhan — APPROVED** [review]

**Pull request merged**